src/misc.jl

```diff
@@ -392,8 +392,8 @@ function trim_zip(xs...)
 end
 
 # Convenience constructors of IndirectArrays
-discretize_make_ia(values::AbstractVector, levels) =
-    IndirectArray(Array{UInt8}(indexin(values, levels)), levels)
+discretize_make_ia(values::AbstractVector, levels::AbstractVector{T}) where T =
+    IndirectArray{T,1,AbstractVector{UInt32},AbstractVector{T}}(indexin(values, levels), levels)
```

Comment:
Shouldn't using UInt work here and letting Julia pick the optimal bitstype?

Reply:
UInt is an alias for UInt64